source/common/quic/client_connection_factory_impl.cc

```diff
@@ -49,8 +48,11 @@ createQuicNetworkConnection(Http::PersistentQuicInfo& info, Event::Dispatcher& d
       info_impl->alarm_factory_, quic::ParsedQuicVersionVector{info_impl->supported_versions_[0]},
       local_addr, dispatcher, nullptr);
   auto& static_info = StaticInfo::get();
+
+  ASSERT(!info_impl->supported_versions_.empty());
+  // QUICHE client session always use the 1st version to start handshake.
   auto ret = std::make_unique<EnvoyQuicClientSession>(
-      static_info.quic_config_, info_impl->supported_versions_, std::move(connection),
+      info_impl->quic_config_, info_impl->supported_versions_, std::move(connection),
```

Comment:
Should the send_buffer_limit be set when constructing this object?  It seems to be set to 0 (e.i. unlimited)

I guess this is the client side instead of the server side; worth adding a TODO for upstream config fields?

Reply:
info_impl->quic_config_ is configured in conn_pool according to host_->cluster().http3Options().quic_protocol_options().

Reply:
I think you missed my question about send_buffer_limit specified by the last argument which is set to 0.